src/wallet/scriptpubkeyman.cpp

```diff
@@ -304,6 +345,19 @@ void LegacyScriptPubKeyMan::MarkUnusedAddresses(const CScript& script)
                 WalletLogPrintf("%s: Topping up keypool failed (locked wallet)\n", __func__);
             }
         }
+
+        auto it = mapKeyMetadata.find(keyid);
+        if (it != mapKeyMetadata.end()){
+            CKeyMetadata meta = it->second;
+            if (!meta.hd_seed_id.IsNull() && meta.hd_seed_id != m_hd_chain.seed_id) {
+                bool internal = meta.key_origin.path[1] - BIP32_HARDENED_KEY_LIMIT != 0;
```

Comment:
nit: it seems slightly inconsistent that we're sometimes using BIP32_HARDENED_KEY_LIMIT as a value for arithmetic operations and sometimes as a bit for bitwise operations. Functionally it's fine, but It seems slightly clearer to me to do:

```
bool internal = meta.key_origin.path[1] & ~BIP32_HARDENED_KEY_LIMIT != 0;
```

and the same below for `index`

Reply:
Done